lib/flowViewport/sceneIndex/fvpLightsManagementSceneIndex.cpp

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright 2024 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//Local headers
+#include "fvpLightsManagementSceneIndex.h"
+
+//USD/Hydra headers
+#include <pxr/imaging/hd/tokens.h>
+#include <pxr/imaging/hd/retainedDataSource.h>
+#include <pxr/imaging/hd/overlayContainerDataSource.h>
+#include <pxr/imaging/hd/containerDataSourceEditor.h>
+#include <pxr/imaging/hd/sceneIndexPrimView.h>
+#include <pxr/imaging/hd/light.h>
+#include <pxr/imaging/hd/lightSchema.h>
+#include <pxr/imaging/glf/simpleLight.h>
+
+//ufe
+#include <ufe/globalSelection.h>
+#include <ufe/observableSelection.h>
+
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace {
+
+bool _IsALight(const TfToken& primType)
+{
+    static const TfTokenVector & lightsToken = HdLightPrimTypeTokens();//Get all light tokens from USD
+    return lightsToken.end() != std::find(lightsToken.begin(), lightsToken.end(), primType);
+}
+
+void _DisableLight(HdSceneIndexPrim& prim)
+{
+    //We don't set the intensity to 0 as for domelights this makes disappear the geometry
+    HdContainerDataSourceEditor editor(prim.dataSource);
+    editor.Set(
+        HdLightSchema::GetDefaultLocator().Append(HdLightTokens->ambient),
+        HdRetainedTypedSampledDataSource<float>::New(0.0f)
+    );
+    editor.Set(
+        HdLightSchema::GetDefaultLocator().Append(HdLightTokens->diffuse),
+        HdRetainedTypedSampledDataSource<float>::New(0.f)
+    );
+    editor.Set(
+        HdLightSchema::GetDefaultLocator().Append(HdLightTokens->specular),
+        HdRetainedTypedSampledDataSource<float>::New(0.f)
+    );
```

Comment:
Minor: would the following work?
```suggestion
    for (const auto& t : {HdLightTokens->ambient, HdLightTokens->diffuse, HdLightTokens->specular}) {
        editor.Set(HdLightSchema::GetDefaultLocator().Append(t),
            HdRetainedTypedSampledDataSource<float>::New(0.0f)
        );
    }
```